tasks/ast_tools/src/derives/estree.rs

```diff
@@ -299,10 +299,16 @@ impl<'s> StructSerializerGenerator<'s> {
             let via_ty = parse_str::<Type>(via_str).unwrap();
             value = quote!( #via_ty::from(&#value) );
         } else if let Some(append_field_index) = field.estree.append_field_index {
-            let append_from_ident = struct_def.fields[append_field_index].ident();
-            value = quote! {
-                AppendTo { array: &#value, after: &#self_path.#append_from_ident }
+            let append_field = &struct_def.fields[append_field_index];
+            let append_from_ident = append_field.ident();
+            let wrapper = if append_field.type_def(self.schema).is_option() {
+                quote! { AppendTo }
+            } else {
+                quote! { oxc_estree::ser::AppendToConcat }
```

Comment:
Style nit: Can we make `AppendTo` and `AppendToConcat` consistent? Either (1) add import of  `AppendToConcat` in `fn prelude` (same as `AppendTo`), or (2) remove `AppendTo` from the prelude, and include both with their full path `oxc_estree::ser::...` here.